lib/libc/minimal/source/time/asctime.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021 Sun Amar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include <time.h>
+#include <bits/restrict.h>
+
+static char *z_impl_asctime(const struct tm *timeptr, char *buf, size_t buflen)
+{
+	static char wday_name[7][3] = {
```

Comment:
`static const char* wday_name[] = {`

The string values should be in ROM.  Then get rid of the precision argument in the sprintf format string.  Or leave it, and switch this to:

    static const char* wdays = "SunMonTueWedThuFriSat???";
    size_t wdidx = ((unsigned int)timeptr->tm_wday < 7U) ? (3U * timeptr->tm_wday) : (3U * 7U);

and use `&wdays[wdidx]`.  See below for range checking.

Whichever you do, do the same for month names.